cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -0,0 +1,90 @@
+set (LZ4_VERSION "1.7.5")
+set (SNAPPY_VERSION "1.1.3")
+set (ZLIB_VERSION "1.2.11")
+set (THIRDPARTY_DIR "${CMAKE_BINARY_DIR}/c++/libs/thirdparty")
+
+string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_BUILD_TYPE)
+
+# ----------------------------------------------------------------------
+# Snappy
+
+set (SNAPPY_PREFIX "${THIRDPARTY_DIR}/snappy_ep-install")
+set (SNAPPY_HOME "${SNAPPY_PREFIX}")
+set (SNAPPY_INCLUDE_DIRS "${SNAPPY_PREFIX}/include")
+set (SNAPPY_STATIC_LIB_NAME snappy)
+set (SNAPPY_STATIC_LIB "${SNAPPY_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}${SNAPPY_STATIC_LIB_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+set (SNAPPY_SRC_URL "${CMAKE_SOURCE_DIR}/c++/libs/snappy-${SNAPPY_VERSION}.tar.gz")
+if (${UPPERCASE_BUILD_TYPE} EQUAL "RELEASE")
+   set (SNAPPY_CXXFLAGS "CXXFLAGS='-DNDEBUG -O2'")
+endif ()
+
+ExternalProject_Add (snappy_ep
+  CONFIGURE_COMMAND ./configure "--prefix=${SNAPPY_PREFIX}" ${SNAPPY_CXXFLAGS}
+  BUILD_IN_SOURCE 1
+  BUILD_COMMAND ${MAKE}
+  INSTALL_DIR ${SNAPPY_PREFIX}
+  URL ${SNAPPY_SRC_URL}
```

Comment:
+1. I mentioned using HTTP URL(download the tar files) as future work because I wasn't sure of its impact on current users. I will update to use URLs in this patch if there are no concerns.